packages/paima-sdk/paima-chain-types/src/types.ts

```diff
@@ -43,6 +43,11 @@ export interface SubmittedData {
     contractAddress: null | string;
     /** Name/id of the primitive that triggered this event, if known */
     primitiveName: null | string;
+
+    /* Timestamp if this is a timer scheduled with a date instead of a block
+     * number. This will be always lower or equal than the underlying block's
+     * timestamp */
+    scheduledAt: null | Date;
```

Comment:
I would specify the unit of time here (ex: `scheduleAtMs` for milliseconds

Reply:
isn't that already implied by the `Date` type though? And it could actually be something else, since you could call `toISOString` on it or something (instead of `getTime()`)

Or should we change this to `number` instead?

Reply:
oh my bad, I didn't notice you used `Date` as the type. Date isn't JSON compatible, so I would avoid it for types that are likely to be converted to JSON like this one, but I'm okay either way